cpp/Platform.Converters/CachingConverterDecorator.h

```diff
@@ -1,15 +1,28 @@
-﻿namespace Platform::Converters
+﻿#pragma once
+#include <functional>
+#include <memory>
+#include <unordered_map>
+
+namespace Platform::Converters
 {
     template <typename ...> class CachingConverterDecorator;
-    template <typename TSource, typename TTarget> class CachingConverterDecorator<TSource, TTarget> : public IConverter<TSource, TTarget>
+    template <typename TSource, typename TTarget> class CachingConverterDecorator<TSource, TTarget>
     {
-        private: readonly IConverter<TSource, TTarget> *_baseConverter;
-        private: readonly IDictionary<TSource, TTarget> *_cache;
+        private: std::unique_ptr<std::function<TTarget(TSource)>> _baseConverter;
```

Comment:
Do not depend directly on `std::function` type here.
Let the user specify any callable type here. 
The is no need to restrict the user with pointer here.
Catching converter can be a container for callable object.

Reply:
Yes. Forward converter as template parameter.
```c++
template<typename F>
struct Converter {
    F base;
    Converter(F&& base) : base(base) {}
}

template<typename F>
Converter(F) -> Converter(F);
```